actionpack/actionpack.gemspec

```diff
@@ -35,7 +35,8 @@ Gem::Specification.new do |s|
 
   s.add_dependency "activesupport", version
 
-  s.add_dependency "rack",      "~> 2.0", ">= 2.2.4"
+  s.add_dependency "rack", ">= 2.2.4", "< 4"
+  s.add_dependency "rack-session"
```

Comment:
How does this work with rack 2? How are we use that the code in rack-session is the one that is being required when we `require "rack/session/abstract/id"? Doesn't it depends on the order those gems are in the load path?

Reply:
The current release of `rack-session` IIRC is essentially line-for-line file-for-file compatible so it's okay to load it with Rack 2. However, I imagine this will eventually change, so the plan is this:

1. Release a stable 1.x release that is compatible with the current interface and should be okay to have loaded with Rack 2.
2. Release a stable 2.x release that is only for Rack >= 3.0 when that interface is changed in any substantial way.